src/main/java/org/cyclonedx/generators/json/BomJsonGenerator12.java

```diff
@@ -38,7 +41,13 @@
      * @param bom the BOM to generate
      */
     public BomJsonGenerator12(final Bom bom) {
-        this.bom = bom;
+        Bom modifiedBom = null;
+        try {
+            modifiedBom = injectBomFormatAndSpecVersion(bom);
+        }
+        catch (GeneratorException e) {
```

Comment:
I don't think a log is helpful but we want to need to handle the exception, in case something happens when generating the bom

Reply:
Yeah there is no current logger (although I didn't think having one would be a bad idea, just didn't want to add to this already huge PR).

@stevespringett so in this case, we catch an exception, how do you want to handle this? I'm gonna think it will never happen (the caller or try in this case is accessing fields on the class, and I can't personally think of an actual case where you'll get the exceptions, but who knows), but in the eventual case where I'm wrong (always am), what would you like the behavior to be?